core/field_label.js

```diff
@@ -38,18 +38,30 @@ goog.require('Blockly.utils.Size');
  * Class for a non-editable, non-serializable text field.
  * @param {string=} opt_value The initial value of the field. Should cast to a
  *    string. Defaults to an empty string if null or undefined.
- * @param {string=} opt_class Optional CSS class for the field's text.
+ * @param {Object=} opt_config A map of options used to configure the field.
+ *    See the [field creation documentation]{@link https://developers.google.com/blockly/guides/create-custom-blocks/fields/built-in-fields/label#creation}
+ *    for a list of properties this parameter supports.
  * @extends {Blockly.Field}
  * @constructor
  */
-Blockly.FieldLabel = function(opt_value, opt_class) {
+Blockly.FieldLabel = function(opt_value, opt_config) {
```

Comment:
I'm not a fan of this. I know the goal here is to move away from setting configuration options in the constructor, and I'm totally okay with that plan moving forward with new fields. Unfortunately due to the legacy nature of these constructors, we don't want to break existing developers.
I recognize we can do that by doing a check on the constructor parameters but for the same reasons that were outlined in FieldImage, the code gets messy very quickly and hard to parse. Here are our options, and I'll see what others think: 
1. Append opt_config to existing fields, for some configuration options like ``class`` above, it might mean there are multiple ways to configure the same option (constructor or property bag), and code like this:
```
new Blockly.FieldLabel(val, 'test');
new Blockly.FieldLabel(val, null, { class: 'test'});
```
Not great for developers using the new config, as they have to skip parameters, but clean in code.
2. Do what you did here, which is check for the value of the parameter, and switch constructor paths based on that.
Not clean in code, compiler types are inconsistent (because now you're saying opt_config can be a string or an Object but you only handle that internally, and don't surface it in jsdoc to developers (docs) and the closure compiler. 
3. We could make handle both constructors, but in a way the compiler understands:
```
/**
 * Class for a non-editable, non-serializable text field.
 * @param {string=} opt_value The initial value of the field. Should cast to a
 *    string. Defaults to an empty string if null or undefined.
 * @param {string|Object=} opt_classOrConfig Either an optional CSS class, or a
 * map of options used to configure the field.
 * @extends {Blockly.Field}
 * @constructor
 */
Blockly.FieldLabel = function (opt_value, opt_classOrConfig) {
  if (typeof opt_classOrConfig == 'string') {
    return Blockly.FieldLabel.constructWithClass_(opt_value,
      /** @type {string} */ opt_classOrConfig);
  } else {
    return Blockly.FieldLabel.constructWithConfig_(opt_value,
      /** @type {Object} */ opt_classOrConfig)
  }
};

/**
 * Class for a non-editable, non-serializable text field.
 * @param {string=} opt_value The initial value of the field. Should cast to a
 *    string. Defaults to an empty string if null or undefined.
 * @param {string=} opt_class Optional CSS class for the field's text.
 * @constructor
 */
Blockly.FieldLabel.constructWithClass_ = function (opt_value, opt_class) {
  ...
};

/**
 * Class for a non-editable, non-serializable text field.
 * @param {string=} opt_value The initial value of the field. Should cast to a
 *    string. Defaults to an empty string if null or undefined.
 * @param {Object=} opt_config A map of options used to configure the field.
 *    See the [field creation documentation]{@link https://developers.google.com/blockly/guides/create-custom-blocks/fields/built-in-fields/label#creation}
 *    for a list of properties this parameter supports.
 * @constructor
 */
Blockly.FieldLabel.constructWithConfig_ = function (opt_value, opt_config) {
  ...
};
```
Notice how the constructor is the one branching out the two constructor paths. It's also type casting the parameter, and each constructor has the correct type description.

@rachel-fenichel @RoboErikG what are your thoughts on dealing with this backwards compat issue with all these fields?